crates/ide-diagnostics/src/handlers/unused_variables.rs

```diff
@@ -23,6 +24,13 @@ pub(crate) fn unused_variables(
         return None;
     }
     let diagnostic_range = ctx.sema.diagnostics_display_range(ast);
+    // The range for the Actual Name. We don't want to replace the entire declarition. Using the diagnostic range causes issues within in Array Destructuring.
+    let name_range =
+        d.local.primary_source(ctx.sema.db).name().map(|v| v.syntax().value.text_range())?;
+    // Make sure we are within the diagnostic range for the variable
+    if !diagnostic_range.range.contains_range(name_range) {
```

Comment:
This check is still wrong, `diagnostic_range` is from a real file where as `name_range` could still originate from a macro expansion.

```rust
let name_range = d.local.primary_source(ctx.sema.db).name().map(|v| v.syntax().clone()).as_ref().original_file_range_rooted(self.db.upcast()).filter(|it| it.file_id == ast.file_id);
```
should give the right thing, and if it's `None` (can happen in theory via proc-macros/includes I believe), we should just not offer the quickfix

Reply:
That is, ideally this function shouldnt ever return `None`, its a temporary fix for another issue. As returning `None` means the diagnostic disappears.